widget/table.go

```diff
@@ -0,0 +1,301 @@
+	} else {
+		offY := t.t.SelectedRow*(t.cellSize.Height+1) - t.scroll.Offset.Y
+		y1 := theme.Padding() + offY
+		y2 := y1 + t.cellSize.Height
+		if y2 < theme.Padding() || y1 > t.t.size.Height {
+			t.rowMarker.Hide()
+		} else {
+			t.rowMarker.Show()
+
+			top := fyne.Max(theme.Padding(), y1)
+			t.rowMarker.Move(fyne.NewPos(0, top))
+			t.rowMarker.Resize(fyne.NewSize(theme.Padding(), fyne.Min(y2, t.t.size.Height)-top))
+		}
+	}
+	t.rowMarker.Refresh()
+
+	colDivs := int(math.Ceil(float64(t.t.size.Width+1) / float64(t.cellSize.Width+1)))   // +1 for div width
+	rowDivs := int(math.Ceil(float64(t.t.size.Height+1) / float64(t.cellSize.Height+1))) // +1 for div width
+
+	if len(t.dividers) < colDivs+rowDivs {
+		for i := len(t.dividers); i < colDivs+rowDivs; i++ {
+			t.dividers = append(t.dividers, canvas.NewRectangle(theme.ShadowColor()))
+		}
+
+		obj := []fyne.CanvasObject{t.scroll, t.colMarker, t.rowMarker}
+		t.objects = append(obj, t.dividers...)
+	}
+
+	divs := 0
+	i := 0
+	rows, cols := t.t.DataSize()
+	for x := theme.Padding() + t.scroll.Offset.X - (t.scroll.Offset.X % (t.cellSize.Width + 1)) - 1; x < t.scroll.Offset.X+t.t.size.Width && i < cols-1; x += t.cellSize.Width + 1 {
+		if x <= theme.Padding()+t.scroll.Offset.X {
+			continue
+		}
+		i++
+
+		t.dividers[divs].Move(fyne.NewPos(x-t.scroll.Offset.X, theme.Padding()))
+		t.dividers[divs].Resize(fyne.NewSize(1, t.t.size.Height-theme.Padding()*2))
+		t.dividers[divs].Show()
+		divs++
+	}
+
+	i = 0
+	for y := theme.Padding() + t.scroll.Offset.Y - (t.scroll.Offset.Y % (t.cellSize.Height + 1)) - 1; y < t.scroll.Offset.Y+t.t.size.Height && i < rows-1; y += t.cellSize.Height + 1 {
+		if y <= theme.Padding()+t.scroll.Offset.Y {
+			continue
+		}
+		i++
+
+		t.dividers[divs].Move(fyne.NewPos(theme.Padding(), y-t.scroll.Offset.Y))
+		t.dividers[divs].Resize(fyne.NewSize(t.t.size.Width-theme.Padding()*2, 1))
+		t.dividers[divs].Show()
+		divs++
+	}
+
+	for i := divs; i < len(t.dividers); i++ {
+		t.dividers[divs].Hide()
+	}
+}
+
+func (t *tableRenderer) Layout(s fyne.Size) {
+	t.moveIndicators()
+
+	t.scroll.Move(fyne.NewPos(theme.Padding(), theme.Padding()))
+	t.scroll.Resize(s.Subtract(fyne.NewSize(theme.Padding(), theme.Padding())))
+}
+
+func (t *tableRenderer) MinSize() fyne.Size {
+	return t.cellSize
+}
+
+func (t *tableRenderer) Refresh() {
+	template := t.t.NewCell()
+	t.cellSize = template.MinSize().Add(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	t.moveIndicators()
+
+	t.colMarker.FillColor = theme.PrimaryColor()
+	t.colMarker.Refresh()
+	t.rowMarker.FillColor = theme.PrimaryColor()
+	t.rowMarker.Refresh()
+
+	for _, div := range t.dividers {
+		div.(*canvas.Rectangle).FillColor = theme.ShadowColor()
+		div.Refresh()
+	}
+	t.t.cells.Refresh()
+}
+
+func (t *tableRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (t *tableRenderer) Objects() []fyne.CanvasObject {
+	return t.objects
+}
+
+func (t *tableRenderer) Destroy() {
+}
```

Comment:
Could baseRenderer be embedded to avoid needing these? (would need to call SetObjects rather than objects field)

Reply:
Isn't that better for static objects renderer? I figured this was doing a lot of object manipulation.

Reply:
`objects` is still a field, but it would be inside the embedded type, and allow 3 methods to be dropped from here. The only difference in the rest of the code would be replacing `r.objects = ...` with `r.SetObjects(...)`